Terminal.Gui/Core/Window.cs

```diff
@@ -200,7 +201,7 @@ public override bool MouseEvent (MouseEvent mouseEvent)
 				if (dragPosition.HasValue) {
 					if (SuperView == null) {
 						Application.Top.SetNeedsDisplay (Frame);
-						Application.Top.Redraw (Frame);
+						Application.Top.Redraw (Bounds);
```

Comment:
Change to  Frame please

Reply:
Redraw should never be called with Frame. Frame is superview-relative. `Redraw` is defined to take view-relative, aka `Bounds`. The bug is in `Toplevel.Redraw` where it calls clear with the bounds and not `this.Frame`.

Reply:
There are some cases where we only need to clear a certain region of the screen. We don't need clear all screen if it only a minor part needed to repaint. So I think that the `Redraw `methods had a region as parameter because it could be any frame dimensions we want to pass to it. So in the image you see that the `TextField `at the bottom and the `StatusBar `disapear and there are not redrawed again while dragging. Without these features it's not worth it developing dragging at all. I think that my original suggestion to resolve this is better.

![background](https://user-images.githubusercontent.com/13117724/83660403-b659e200-a5bc-11ea-8ced-e07a82bcc325.gif)

Reply:
Your suggestion is to redefine how `Redraw` is defined. We should do that very carefully. I'm open to doing that because I think the current design *is* suspect. Let's start another Issue to track and debate. Ok?